e2e/tests/core/02-client/client_test.go

```diff
@@ -242,6 +253,57 @@ func (s *ClientTestSuite) TestClient_Update_Misbehaviour() {
 	})
 }
 
+// TestAllowedClientsParam tests changing the AllowedClients parameter using a governance proposal
+func (s *ClientTestSuite) TestAllowedClientsParam() {
+	t := s.T()
+	ctx := context.TODO()
+
+	_, _ = s.SetupChainsRelayerAndChannel(ctx, s.TransferChannelOptions())
+	chainA, chainB := s.GetChains()
+	chainAVersion := chainA.Config().Images[0].Version
+
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 1, chainA, chainB), "failed to wait for blocks")
+
+	t.Run("ensure allowed clients are set to the default", func(t *testing.T) {
+		allowedClients := s.QueryAllowedClients(ctx, chainA)
+		s.Require().Equal(clienttypes.DefaultAllowedClients, allowedClients)
+	})
+
+	allowedClient := ibcexported.Solomachine
+	t.Run("change the allowed client to only allow localhost clients", func(t *testing.T) {
+		if testvalues.SelfParamsFeatureReleases.IsSupported(chainAVersion) {
+			authority, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+			s.Require().NoError(err)
+			s.Require().NotNil(authority)
+
+			msg := clienttypes.NewMsgUpdateParams(authority.String(), clienttypes.NewParams(allowedClient))
+			s.ExecuteGovProposalV1(ctx, msg, chainA, chainAWallet, 1)
+		} else {
+			value, err := tmjson.Marshal([]string{allowedClient})
+			s.Require().NoError(err)
+			changes := []paramsproposaltypes.ParamChange{
+				paramsproposaltypes.NewParamChange(ibcexported.ModuleName, string(clienttypes.KeyAllowedClients), string(value)),
+			}
+
+			proposal := paramsproposaltypes.NewParameterChangeProposal(ibctesting.Title, ibctesting.Description, changes)
+			s.ExecuteGovProposal(ctx, chainA, chainAWallet, proposal)
+		}
+	})
+
+	t.Run("validate the param was successfully changed", func(t *testing.T) {
+		allowedClients := s.QueryAllowedClients(ctx, chainA)
+		s.Require().Equal([]string{allowedClient}, allowedClients)
```

Comment:
nit: we could maybe use `s.Require().Contains()` here, makes no difference to me